nodejs build style (#23947)
void-linux/void-packages: Manual.md

dependencies to the root directory, and only checks if a binary package is
available in repositories.

> - `nodejs` For apps made in Node.js that get built with npm or yarn

Reviewer: I think the explanation of variables could be better formatted as a list.  Look at the python section starting on line 1504.

- `make_build_args`: This is passed as-is to `stack build ...`, so
  you can add your `--flag ...` parameters there.

> <a id="pkgs_nodejs"></a>

Reviewer 2: This needs to go on the table of contents at the start of the document.

### Node.js packages

Node.js packages should be built with the `nodejs` build style, if possible.
The `nodejs` build style takes care of downloading yarn/npm dependencies
and running builds via package.json scripts.

The following template variables influence how Node.js packages are built:
- `nodejs_build_script`: the package.json script that builds the app files
	(default: `build`).

> - `nodejs_install_script`: app installing script, which most probably 
> 	doesn't even exist and you should redefine `do_install`
> 	in your build script. 

Reviewer 2: ```suggestion
- `nodejs_install_script`: app installing script, which is unlikely to exist.
Most of the time, you will have to define your own `do_install`.
```

- `nodejs_packager`: whether the build script should use yarn or npm.
	Set to `yarn` if the `yarn.lock` file exists, else it's set to `npm`

> - `NODE_ENV`: set to `production` by default

Reviewer 2: it should be exported in the build style, otherwise just setting it won't affect the build.
